connection/connection_test.go

```diff
@@ -380,6 +391,10 @@ func TestExplicitReconnect(t *testing.T) {
 }
 
 func TestConnectMetrics(t *testing.T) {
+	contextFromKtesting := func() context.Context {
+		_, ctx := ktesting.NewTestContext(t)
+		return ctx
+	}
```

Comment:
This is unnecessary: all test cases get a new Context, but it's all for the same testing.T. You get the same effect by creating just one Context here and then using it for each test case.

If this was using sub-tests, then sharing the same context would not enable proper per-sub-test output. But it doesn't. so a single context is fine.